Userland/Libraries/LibAccelGfx/Context.h

```diff
@@ -22,17 +24,25 @@ class Context {
 
     static OwnPtr<Context> create();
 
+#ifdef AK_OS_MACOS
+    Context()
+    {
+    }
+#else
     Context(EGLDisplay egl_display, EGLContext egl_context, EGLConfig egl_config)
         : m_egl_display(egl_display)
         , m_egl_context(egl_context)
         , m_egl_config(egl_config)
     {
     }
+#endif
 
 private:
+#ifndef AK_OS_MACOS
```

Comment:
Instead of doing ifdefs in this header for the class details and ctor implementation, can you make use of the PIMPL idiom?

Reply:
yeah, actually I found that we can use default constructor on both platforms because EGLDisplay, EGLConfig are no longer need to be saved after we got rid of PBuffers usage.